superset/explore/form_data/commands/update.py

```diff
@@ -64,7 +65,7 @@ def run(self) -> Optional[str]:
                 # Generate a new key if tab_id changes or equals 0
                 tab_id = self._cmd_params.tab_id
                 contextual_key = cache_key(
-                    session.get("_id"), tab_id, dataset_id, chart_id
+                    session.get("_id"), tab_id, datasource_id, chart_id, datasource_type
```

Comment:
It will miss the cache here for old keys because they were generated without the `datasource_type`. We can add logic to also query for the old format or assume that new keys will be created and think about how to clean the old entries.

Reply:
This is creating a new key, though, right?

Reply:
a cache key is just a uuid, right? It's not encrypted data?